### PR TITLE
45 get results by run endpoint

### LINKED ISSUE #45: Update the client and API to take advantage of the TestRail `get_results_for_run` endpoint
The `testrail-python` code currently only utilizes one of the available end points for collecting results:
(from client.py)
```
    # Result Methods
    def results(self, test_id):
        return map(Result, self.api.results(test_id))
```

(from api.py)
```
    # Result Requests
    def results(self, test_id):
        if self._refresh(self._results[test_id]['ts']):
            _results = self._get('get_results/%s' % test_id)
            self._results[test_id]['value'] = _results
            self._results[test_id]['ts'] = datetime.now()
        return self._results[test_id]['value']
```

This is very inefficient if you are trying to collect all results across a series of runs. TestRail offers a `get_results_for_run` endpoint that can return bulk results per run:
http://docs.gurock.com/testrail-api2/reference-results#get_results_for_run

Update the client and API to accept either a test or run, and return results accordingly.

### PR DESCRIPTION
**Pull request by levi-rs**
Closes #45 
Update Results query to work with a Run or individual Test.

### CODE REVIEW

File: testrail/client.py [hunk 1]
```diff
@@ -256,18 +256,22 @@ def _test_by_id(self, test_id, run):
...
+        return ResultContainer(list(map(Result, self.api.results_by_test(test.id))))
 
     @methdispatch
     def result(self):
         return Result()
 
-    @result.register(int)
-    @singleresult
-    def _result_by_id(self, result_id):
-        return filter(lambda r: r.id == result_id, self.results())
```
Comment (levi-rs) [inline testrail/client.py]: Note that I'm not sure this ever worked, as the original `results` method always required a test id. Additionally, results can only be queried if you supply a run, test or run/case combination. They can't be queried in bulk.

**Comment by levi-rs** [comment]
Not sure yet why that one unit test is failing. Everything passes when I run tox locally.

**Comment by levi-rs** [comment]
WOOO, fixed. I was branched off my local master, from before I merged in upstream/master. Should be good now.

**Comment by travispavek** [comment]
Nice work!

**Pull request merged**